### ISSUE TITLE
Why is the mean molecular weight not consistent between all tests?

### ISSUE THREAD

**Issue by psharda** [quokka-astro/quokka#325]
Most tests that use `EOSTraits mean_molecular_weight` define it as a mass quantity, with units of mass. However, three tests (RadPulse, RadStreaming, RadhydroShock) define it as a dimensionless number of order unity. Why? 

```
ps3459@gadi-login-04:/g/data1b/jh2/ps3459/quokka/src>grep -rnw . -e 'mean_molecular_weight'
./RadhydroShell/test_radhydro_shell.cpp:48:	static constexpr double mean_molecular_weight = 2.2 * m_H;
./RadStreaming/test_radiation_streaming.cpp:26:	static constexpr double mean_molecular_weight = 1.0;
./RayleighTaylor3D/test_hydro3d_rt.cpp:27:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./HydroSMS/test_hydro_sms.cpp:26:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./HydroContact/test_hydro_contact.cpp:26:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadhydroShock/test_radhydro_shock.cpp:65:	static constexpr double mean_molecular_weight = mu;
./FCQuantities/test_fc_quantities.cpp:30:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadPulse/test_radiation_pulse.cpp:30:	static constexpr double mean_molecular_weight = 1.0;
./ODEIntegration/test_ode.cpp:19:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./PassiveScalar/test_scalars.cpp:28:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./PrimordialChem/test_primordial_chem.cpp:46:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadhydroShockCGS/test_radhydro_shock_cgs.cpp:68:	static constexpr double mean_molecular_weight = m_H;
./HydroLeblanc/test_hydro_leblanc.cpp:33:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadTube/test_radiation_tube.cpp:40:	static constexpr double mean_molecular_weight = mu;
./RadMarshakAsymptotic/test_radiation_marshak_asymptotic.cpp:29:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadBeam/test_radiation_beam.cpp:36:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./HydroWave/test_hydro_wave.cpp:26:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadTophat/test_radiation_tophat.cpp:41:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./RadShadow/test_radiation_shadow.cpp:37:	static constexpr double mean_molecular_weight = 10. * quokka::hydrogen_mass_cgs;
./HydroShuOsher/test_hydro_shuosher.cpp:28:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
./Cooling/test_cooling.cpp:38:	static constexpr double mean_molecular_weight = quokka::hydrogen_mass_cgs;
```

**Comment by BenWibking** [comment]
It is consistent. Those tests use dimensionless variables for all quantities, not just the mean molecular weight. The mass unit in those tests is 1.

**Comment by psharda** [comment]
But, then how do we do this: `We should be able to just divide out the mass unit when passing it to Microphysics.` as you mentioned in the amrex Slack? We wanted to just divide out the mass unit in EOS.hpp before the call to microphysics eos, but this would also apply to these three tests no

**Comment by BenWibking** [comment]
I think the way to do it is to add a `mass_unit` variable that gives the unit mass in grams. For the three tests with dimensionless variables, we can then set this to the atomic mass unit in grams.

**Comment by psharda** [comment]
Eh? I didn't follow the second part. Also, where should this unit be defined?

**Comment by BenWibking** [comment]
I think the place to put it is in `EOS_Traits`.

I think there are two sensible options: i) the `mass_unit` is defined to be the unit mass in grams used by the problem's system of units, or ii) `mass_unit` is defined to be the value of the atomic mass unit in the problem's system of units. The first option has the advantage that it is conceptually clear to the reader, and how systems of units are usually defined in various codes. The second option has the advantage that if the problem uses dimensionless units, then it can just be set to `1` and everything will work.

What I meant by my earlier comment is that for option i), then problems that use dimensionless units will have to set their `mass_unit` to the value of the atomic mass unit in grams for all of the dimensional factors to cancel out.

**Comment by psharda** [comment]
Will be fixed by #324
